Updates to eight schools data and notebooks

sethaxen opened the pull request:
Currently non-centered eight schools is missing the `school` dimension on `obs` in `observed_data`, `XXX_predictive`, and `log_likelihood`, instead using the default `obs_dim_0`. centered eight schools, on the other hand, has the `school` dim. This PR adds the missing school dim to non-centered eight schools.

EDIT: It also removes `scores` (which is identical to `obs`) from `constant_data` and add `sigma` to `constant_data` for both centered and non-centered models.

sethaxen committed:
Update non-centered eight schools notebook

sethaxen committed:
Update non-centered eight schools data

sethaxen commented:
After this PR is merged, we should tag a new release. Since we don't really have a set of rules for our release numbers, I'd recommend v0.3.0.

sethaxen commented:
I've kept `scores` in `constant_data`, but I personally don't think it should be in there, since it's identical to `obs` in `observed_data`. Instead, perhaps `sigma` should be in `constant_data`.

aloctavodia commented:
It makes sense to add `sigma` as `constant_data`. I do not have a strong opinion about keeping or removing `scores`

sethaxen committed:
Remove scores and add sigma to constant data

sethaxen committed:
Update centered_eight to match non_centered_eight

sethaxen commented:
> It makes sense to add `sigma` as `constant_data`. I do not have a strong opinion about keeping or removing `scores`

I added `sigma` and removed `scores`.
